tests/schema/test_resolvers.py

```diff
@@ -207,7 +208,7 @@ class User:
         age: int
 
         @classmethod
-        def get_users(cls):
+        def get_users(cls) -> "List[User]":
```

Comment:
we need to add tests to make sure that we use the type defined on the class, and that we throw when the types are different (between resolvers and the class field) :)

Reply:
How strict do we want to be? Should any of these be valid?
```python
@strawberry.type
class Query:

    @strawberry.interface
    class Cheese:
        name: str
    @strawberry.type
    class Swiss(Cheese):
        canton: str

    def int_resolver() -> int: ...
    def swiss_resolver() -> Swiss: ...
    def uuid_resolver() -> UUID: ...
    
    my_float: float = strawberry.field(resolver=int_resolver)
    my_cheese: Cheese = strawberry.field(resolver=swiss_resolver)
    my_id: strawberry.ID = strawberry.field(resolver=uuid_resolver)
```

I think we already discussed the float/int example and determined that it should throw an exception. What about the others?

Reply:
Discussed on Discord. Current plan is to do the type matching verification in another PR. Will start by enforcing _exact_ matches, potentially being more lenient afterwards.